python/pyarrow/tests/test_extension_type.py

```diff
@@ -1926,3 +1926,31 @@ def test_bool8_scalar():
     assert pa.scalar(1, type=pa.bool8()).as_py() is True
     assert pa.scalar(2, type=pa.bool8()).as_py() is True
     assert pa.scalar(None, type=pa.bool8()).as_py() is None
+
+
+@pytest.mark.parametrize("str_type", (pa.utf8, pa.large_utf8))
+def test_json(str_type):
+    storage_type = str_type()
+    data = ['{"a": 1}', '{"b": 2}', None]
+    storage = pa.array(data, type=storage_type)
+    json_type = pa.json(storage_type)
+
+    assert json_type.extension_name == "arrow.json"
+    assert json_type.storage_type == storage_type
+    assert json_type.__class__ is pa.JsonType
+
+    array = pa.ExtensionArray.from_storage(json_type, storage)
+
+    assert array.to_pylist() == data
+    assert array[0].as_py() == data[0]
+    assert array[2].as_py() is None
+
+    buf = ipc_write_batch(pa.RecordBatch.from_arrays([array], ["json"]))
```

Comment:
Can you also add a pickle roundtrip of the type? (see eg test_opaque_type above)

Reply:
And maybe also a basic cast (again see the structure of `test_opaque_type` and `test_bool8_type` above, you can follow that I think)

Reply:
Added pickle and cast tests. Cast test fails for `extenstion.cast(string_view)`. I've not checked why yet but probably we're missing a cast kernel for view types?
```python
pyarrow.lib.ArrowNotImplementedError: Unsupported cast to string_view from extension<arrow.json>
```

Reply:
Yes, I think casting for string view is not yet implemented

Reply:
I suppose #39634 covers casting.